lib/ansible/module_utils/ec2.py

```diff
@@ -728,3 +728,54 @@ def compare_aws_tags(current_tags_dict, new_tags_dict, purge_tags=True):
             tag_key_value_pairs_to_set[key] = new_tags_dict[key]
 
     return tag_key_value_pairs_to_set, tag_keys_to_unset
+
+
+def manage_tags(resource, resource_id, new_tags, purge_tags, connection, check_mode):
+    """
+        Align the tags on the resource to the desired tags specified in new_tags.  This method will add or optionally
+        delete tags to bring the resources tags into compliance with new_tags.  This method supports check_mode.
+
+        Note: Currently supports tags kept in the resource as "Tags" or "TagSet" - since AWS seems vary where it stores
```

Comment:
I think the contents of this function better exist in the module in question as it's a little overly specific and isn't deduplicating code in other modules. As you've noticed, AWS varies the key in the resource as well as the keys for the tags used in compare_tags, and also the methods themselves for creating/removing tags.

Reply:
I think I agree with you on the potentially overly specific for module_utils.  That being said, this is the second module in a row I've worked on that needed this function and it was copied from a third.  The other two functions are ec2_instance and ec2_eni.  I held off changing those modules as part of this pull request just to minimize scope a bit.

I'm happy to perform either of:
1) Move manage_tags back to ec2_eip
2) De-dupe the redundant code in ec2_instance.  The ec2_eni module will need to be de-duped after https://github.com/ansible/ansible/pull/54950 gets in.

Let me know if you still prefer 1.

Reply:
+1 for thinking about the bigger picture. I agree about minimizing scope for an easier time reviewing/merging things. I think 2) would make a good follow up PR because it probably deserves a more rigorous review than the individual module and it makes more sense to create the utility when it's deduplicating code. It could cause similar problems that boto3_tag_list_to_ansible_dict has had, with people assuming it will handle the inconsistencies in AWS, using it in their module without careful consideration, and having tags be entirely broken. A more specific function name or more flexible handling for other use cases may help with that, since `manage_tags` sounds like it would be generic. Guidelines should probably be added to lib/ansible/modules/cloud/amazon/GUIDELINES.md for when the function is appropriate to use as well.

Reply:
Thanks for the input on this @s-hertel.  I think you're right on the scope of discussion.  I'm going to move manage_tags back into ec2_eip.py for this PR, to limit the scope of the discussion to just this module.  Then I will put together a separate PR that de-dupes manage_tags and moves it to module_utils.